File: ciam/src/main/java/com/crm/ciam/controller/JwksController.java
package com.crm.ciam.controller;

import java.security.KeyPair;
import java.security.interfaces.RSAPublicKey;
import java.util.Base64;
import java.util.List;
import java.util.Map;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class JwksController {

    private final RSAPublicKey publicKey;

    public JwksController(KeyPair rsaKeyPair) {
        this.publicKey = (RSAPublicKey) rsaKeyPair.getPublic();
    }

    @GetMapping("/.well-known/jwks.json")
    public Map<String, Object> jwks() {
        Map<String, Object> jwk = Map.of(
                "kty", "RSA",
                "use", "sig",
                "alg", "RS256",
                "n", Base64.getUrlEncoder().withoutPadding().encodeToString(publicKey.getModulus().toByteArray()),
                "e", Base64.getUrlEncoder().withoutPadding().encodeToString(publicKey.getPublicExponent().toByteArray())
        );
        return Map.of("keys", List.of(jwk));
    }
}

File: ciam/src/main/kotlin/com/crm/ciam/controller/JwksController.kt
package com.crm.ciam.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import java.security.KeyPair
import java.security.interfaces.RSAPublicKey
import java.util.Base64

@RestController
class JwksController(rsaKeyPair: KeyPair) {

    private val publicKey: RSAPublicKey = rsaKeyPair.public as RSAPublicKey

    @GetMapping("/.well-known/jwks.json")
    fun jwks(): Map<String, Any> {
        val jwk = mapOf(
            "kty" to "RSA",
            "use" to "sig",
            "alg" to "RS256",
            "n" to Base64.getUrlEncoder().withoutPadding().encodeToString(publicKey.modulus.toByteArray()),
            "e" to Base64.getUrlEncoder().withoutPadding().encodeToString(publicKey.publicExponent.toByteArray())
        )
        return mapOf("keys" to listOf(jwk))
    }
}
